From c4e099fa32c81e09cca7f5252b8c8cc413d1a056 Mon Sep 17 00:00:00 2001
From: root <vadimp@mellanox.com>
Date: Mon, 16 Mar 2020 02:30:11 +0000
Subject: [PATCH 2/2] config: mellanox configuration

It adds the next configuration flags, used by Mellanox systems:

CONFIG_MLXSW_CORE=m
CONFIG_MLXSW_CORE_HWMON=y
CONFIG_MLXSW_CORE_THERMAL=y
CONFIG_MLXSW_CORE_QSFP=y
CONFIG_MLXSW_PCI=m
CONFIG_MLXSW_I2C=m
CONFIG_MLXSW_MINIMAL=m
CONFIG_I2C_MUX_REG=y
CONFIG_I2C_MUX_MLXCPLD=m
CONFIG_I2C_MLXCPLD=m
CONFIG_SENSORS_IIO_HWMON=y
CONFIG_PMBUS=y
CONFIG_SENSORS_PMBUS=y
CONFIG_SENSORS_LM25066=m
CONFIG_SENSORS_TPS53679=m
CONFIG_SENSORS_UCD9000=m
CONFIG_SENSORS_UCD9200=m
CONFIG_LEDS_MLXREG=m
CONFIG_MLX_PLATFORM=m
CONFIG_MELLANOX_PLATFORM=y
CONFIG_MLXREG_HOTPLUG=m
CONFIG_MLXREG_IO=m
CONFIG_MAX1363=y

Signed-off-by: Vadim Pastrenak <vadimp@melanox.com>
Signed-off-by: Stephen <stephens@mellanox.com>

config: mellanox fan configuration It adds configuration flag, used by Mellanox systems:

SENSORS_MLXREG_FAN=m

Signed-off-by: Oleksandr Shamray <oleksandrs@melanox.com>
Signed-off-by: Stephen <stephens@mellanox.com>

update kernel config add CONFIG_MLX_WDT

Signed-off-by: Mykola Kostenok <c_mykolak@mellanox.com>
Signed-off-by: Stephen <stephens@mellanox.com>

sonic update kernel config mlsxw pci

Signed-off-by: Vadim Pasternak <vadimp@mellanox.com>
Signed-off-by: Stephen <stephens@mellanox.com>

Adjust configurations

Signed-off-by: Stephen <stephens@mellanox.com>
---
 debian/build/build_amd64_none_amd64/.config | 53 +++++++++++++++++++++--------
 1 file changed, 39 insertions(+), 14 deletions(-)

diff --git a/debian/build/build_amd64_none_amd64/.config b/debian/build/build_amd64_none_amd64/.config
index b62b167..d74909b 100644
--- a/debian/build/build_amd64_none_amd64/.config
+++ b/debian/build/build_amd64_none_amd64/.config
@@ -2235,9 +2235,9 @@ CONFIG_EEPROM_LEGACY=m
 CONFIG_EEPROM_MAX6875=m
 CONFIG_EEPROM_93CX6=m
 # CONFIG_EEPROM_93XX46 is not set
+# CONFIG_EEPROM_IDT_89HPESX is not set
 CONFIG_EEPROM_SFF_8436=m
 CONFIG_EEPROM_OPTOE=m
-# CONFIG_EEPROM_IDT_89HPESX is not set
 CONFIG_CB710_CORE=m
 # CONFIG_CB710_DEBUG is not set
 CONFIG_CB710_DEBUG_ASSUMPTIONS=y
@@ -2860,7 +2860,13 @@ CONFIG_MLX5_MPFS=y
 CONFIG_MLX5_CORE_EN_DCB=y
 CONFIG_MLX5_CORE_IPOIB=y
 # CONFIG_MLX5_EN_IPSEC is not set
-# CONFIG_MLXSW_CORE is not set
+CONFIG_MLXSW_CORE=m
+CONFIG_MLXSW_CORE_HWMON=y
+CONFIG_MLXSW_CORE_THERMAL=y
+CONFIG_MLXSW_CORE_QSFP=y
+# CONFIG_MLXSW_PCI is not set
+CONFIG_MLXSW_I2C=m
+CONFIG_MLXSW_MINIMAL=m
 CONFIG_MLXFW=m
 CONFIG_NET_VENDOR_MICREL=y
 # CONFIG_KS8842 is not set
@@ -3874,8 +3880,8 @@ CONFIG_I2C_MUX_GPIO=m
 # CONFIG_I2C_MUX_LTC4306 is not set
 # CONFIG_I2C_MUX_PCA9541 is not set
 CONFIG_I2C_MUX_PCA954x=m
-# CONFIG_I2C_MUX_REG is not set
-# CONFIG_I2C_MUX_MLXCPLD is not set
+CONFIG_I2C_MUX_REG=m
+CONFIG_I2C_MUX_MLXCPLD=m
 CONFIG_I2C_HELPER_AUTO=y
 CONFIG_I2C_SMBUS=m
 CONFIG_I2C_ALGOBIT=m
@@ -3944,7 +3950,7 @@ CONFIG_I2C_VIPERBOARD=m
 #
 # Other I2C/SMBus bus drivers
 #
-# CONFIG_I2C_MLXCPLD is not set
+CONFIG_I2C_MLXCPLD=m
 CONFIG_I2C_STUB=m
 # CONFIG_I2C_SLAVE is not set
 # CONFIG_I2C_DEBUG_CORE is not set
@@ -4214,7 +4218,7 @@ CONFIG_SENSORS_G760A=m
 # CONFIG_SENSORS_HIH6130 is not set
 CONFIG_SENSORS_IBMAEM=m
 CONFIG_SENSORS_IBMPEX=m
-# CONFIG_SENSORS_IIO_HWMON is not set
+CONFIG_SENSORS_IIO_HWMON=m
 CONFIG_SENSORS_I5500=m
 CONFIG_SENSORS_CORETEMP=m
 CONFIG_SENSORS_IT87=m
@@ -4243,6 +4247,7 @@ CONFIG_SENSORS_MAX6620=m
 CONFIG_SENSORS_MAX6697=m
 CONFIG_SENSORS_MAX31790=m
 # CONFIG_SENSORS_MCP3021 is not set
+CONFIG_SENSORS_MLXREG_FAN=m
 # CONFIG_SENSORS_TC654 is not set
 CONFIG_SENSORS_MENF21BMC_HWMON=m
 CONFIG_SENSORS_ADCXX=m
@@ -4271,7 +4276,24 @@ CONFIG_SENSORS_NCT7802=m
 CONFIG_SENSORS_NCT7904=m
 CONFIG_SENSORS_NPCM7XX=m
 CONFIG_SENSORS_PCF8591=m
-# CONFIG_PMBUS is not set
+CONFIG_PMBUS=m
+CONFIG_SENSORS_PMBUS=m
+# CONFIG_SENSORS_ADM1275 is not set
+# CONFIG_SENSORS_IBM_CFFPS is not set
+# CONFIG_SENSORS_IR35221 is not set
+CONFIG_SENSORS_LM25066=m
+# CONFIG_SENSORS_LTC2978 is not set
+# CONFIG_SENSORS_LTC3815 is not set
+# CONFIG_SENSORS_MAX16064 is not set
+# CONFIG_SENSORS_MAX20751 is not set
+# CONFIG_SENSORS_MAX31785 is not set
+# CONFIG_SENSORS_MAX34440 is not set
+# CONFIG_SENSORS_MAX8688 is not set
+# CONFIG_SENSORS_TPS40422 is not set
+CONFIG_SENSORS_TPS53679=m
+CONFIG_SENSORS_UCD9000=m
+CONFIG_SENSORS_UCD9200=m
+# CONFIG_SENSORS_ZL6100 is not set
 # CONFIG_SENSORS_SHT15 is not set
 CONFIG_SENSORS_SHT21=m
 # CONFIG_SENSORS_SHT3x is not set
@@ -4372,6 +4394,7 @@ CONFIG_MENF21BMC_WATCHDOG=m
 CONFIG_WDAT_WDT=m
 # CONFIG_XILINX_WATCHDOG is not set
 # CONFIG_ZIIRAVE_WATCHDOG is not set
+CONFIG_MLX_WDT=y
 # CONFIG_CADENCE_WATCHDOG is not set
 # CONFIG_DW_WATCHDOG is not set
 # CONFIG_MAX63XX_WATCHDOG is not set
@@ -6466,7 +6489,7 @@ CONFIG_LEDS_MENF21BMC=m
 #
 # CONFIG_LEDS_BLINKM is not set
 # CONFIG_LEDS_MLXCPLD is not set
-# CONFIG_LEDS_MLXREG is not set
+CONFIG_LEDS_MLXREG=m
 # CONFIG_LEDS_USER is not set
 # CONFIG_LEDS_NIC78BX is not set
 
@@ -6674,7 +6697,7 @@ CONFIG_INTEL_MIC_X100_DMA=m
 # CONFIG_QCOM_HIDMA is not set
 CONFIG_DW_DMAC_CORE=m
 CONFIG_DW_DMAC=m
-# CONFIG_DW_DMAC_PCI is not set
+CONFIG_DW_DMAC_PCI=m
 CONFIG_HSU_DMA=y
 
 #
@@ -7055,7 +7078,7 @@ CONFIG_INTEL_PMC_IPC=m
 CONFIG_SURFACE_PRO3_BUTTON=m
 # CONFIG_SURFACE_3_BUTTON is not set
 # CONFIG_INTEL_PUNIT_IPC is not set
-# CONFIG_MLX_PLATFORM is not set
+CONFIG_MLX_PLATFORM=m
 # CONFIG_INTEL_TURBO_MAX_3 is not set
 # CONFIG_INTEL_CHTDC_TI_PWRBTN is not set
 # CONFIG_I2C_MULTI_INSTANTIATE is not set
@@ -7066,7 +7089,9 @@ CONFIG_CHROMEOS_LAPTOP=m
 CONFIG_CHROMEOS_PSTORE=m
 # CONFIG_CHROMEOS_TBMC is not set
 CONFIG_CROS_KBD_LED_BACKLIGHT=m
-# CONFIG_MELLANOX_PLATFORM is not set
+CONFIG_MELLANOX_PLATFORM=y
+CONFIG_MLXREG_HOTPLUG=m
+CONFIG_MLXREG_IO=m
 CONFIG_CLKDEV_LOOKUP=y
 CONFIG_HAVE_CLK_PREPARE=y
 CONFIG_COMMON_CLK=y
@@ -7252,7 +7277,7 @@ CONFIG_MMA9553=m
 # CONFIG_MAX1027 is not set
 # CONFIG_MAX11100 is not set
 # CONFIG_MAX1118 is not set
-# CONFIG_MAX1363 is not set
+CONFIG_MAX1363=m
 # CONFIG_MAX9611 is not set
 # CONFIG_MCP320X is not set
 # CONFIG_MCP3422 is not set
-- 
1.9.1

